src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -19,93 +20,253 @@
-					{{ item.utils.counter }}
-				</AppNavigationCounter>
-
-				<!-- first action if only one action -->
-				<li v-if="item.utils.actions && item.utils.actions.length === 1"
-					class="app-navigation-entry-utils-menu-button">
-					<button :class="item.utils.actions[0].icon" :title="item.utils.actions[0].text" @click="item.utils.actions[0].action" />
-				</li>
-
-				<!-- menu button if at least two actions -->
-				<li v-else-if="item.utils.actions && item.utils.actions.length > 1"
-					class="app-navigation-entry-utils-menu-button">
-					<button v-click-outside="hideMenu" @click="toggleMenu" />
-				</li>
-			</ul>
-		</div>
+# Usage
 
-		<!-- menu if at least two actions -->
-		<div v-if="item.utils && item.utils.actions && item.utils.actions.length > 1"
-			:class="{ 'open': openedMenu }" class="app-navigation-entry-menu">
-			<PopoverMenu :menu="item.utils.actions" />
-		</div>
+### Simple element
+
+* With an icon:
+
+```
+<AppNavigationItem title="My title" icon="icon-category-enabled">
+
+```
+* With a spinning loader instead of the icon:
+
+<AppNavigationItem title="Loading Item" :loading="true" />
+
+### Element with actions
+Wrap the children in a template. If you have more than 2 actions, a popover menu and a menu
+button will be automatically created.
+
+```
+<AppNavigationItem title="Item with actions" icon="icon-category-enabled">
+	<template slot="actions">
+		<ActionButton icon="icon-edit" @click="alert('Edit')">
+			Edit
+		</ActionButton>
+		<ActionButton icon="icon-delete" @click="alert('Delete')">
+			Delete
+		</ActionButton>
+		<ActionLink icon="icon-external" title="Link" href="https://nextcloud.com" />
+	</template>
+</AppNavigationItem>
+```
+
+### Element with counter
+Just nest the counter into <AppNavigationItem> and add `slot="counter"` to it.
+
+```
+<AppNavigationItem title="Item with counter" icon="icon-folder">
+	<AppNavigationCounter slot="counter">
+		99+
+	</AppNavigationCounter>
+</AppNavigationItem>
+```
+
+### Element with children
+
+Wrap the children in a template with the `slot` property and use the prop `allowCollapse` to choose wether to allow or
+prevent the user from collapsing the items.
+
+```
+<AppNavigationItem title="Item with children" :allowCollapse="true">
+	<template slot="children">
+		<AppNavigationItem title="AppNavigationItemChild1" />
+		<AppNavigationItem title="AppNavigationItemChild2" />
+		<AppNavigationItem title="AppNavigationItemChild3"  />
+		<AppNavigationItem title="AppNavigationItemChild4"  />
+	</template>
+</AppNavigationItem>
+```
+### Editable element
+Add the prop `:editable=true` and an edit placeholder if you need it. By devault
+the placeholder is the previous title of the element.
+
+```
+<AppNavigationItem title="Editable Item" :editable="true"
+	editPlaceholder="your_placeholder_here" />
+```
+### Undo element
+Just set the `undo` and `undo-text` props.
+
+```
+<AppNavigationItem :undo="true" undo-text="Are you sure you want to delete?"  />
```

Comment:
As far as I can see, `undo-text` isn't also used anymore. You have to use `title`, hmm?

Reply:
Nope, seems like we lost this during a rebase :thinking:

Reply:
Yep, my bad @ma12-co https://github.com/nextcloud/nextcloud-vue/commit/1620213d2fed461d72e435de22e6fb8f1883b88a

Reply:
Yes, I removed it and forgot to update the docs! Thanks :)

Reply:
>Yep, my bad @ma12-co 

No, I removed it on purpose, I decided to just leave title prop only

Reply:
Ah! I thought I did a mistake! :+1: